src/Views/Form/Templates/Classic/resources/js/form.js

```diff
@@ -246,9 +246,8 @@ function moveDefaultGatewayDataIntoActiveGatewaySection() {
 
     addSelectedGatewayDetails(createGatewayDetails());
 
-    document
-        .querySelector('.give-gateway-details')
-        .append(...document.querySelectorAll('#give_purchase_form_wrap fieldset:not(.give-donation-submit)'));
+    document.querySelector('#give_purchase_form_wrap').removeChild(document.querySelector('.give-donation-submit'));
+    document.querySelector('.give-gateway-details').append(...document.querySelector('#give_purchase_form_wrap').children);
 
     removeNode(document.querySelector('#give_purchase_form_wrap'));
```

Comment:
That's a good approach. In case you think it's relevant, we can also optimize the code a bit avoiding unnecessary use of `document.querySelector()` by storing the reference to the `#give_purchase_form_wrap` element in a variable:

```suggestion
    const purchaseFormWrap = document.querySelector('#give_purchase_form_wrap');
    purchaseFormWrap.removeChild(purchaseFormWrap.querySelector('.give-donation-submit'));
    document.querySelector('.give-gateway-details').append(...purchaseFormWrap.children);

    removeNode(purchaseFormWrap);
```

Reply:
Resolved in dd3165a8122493ce0b6d7d082e34f268e48e3ed7